### PR TITLE
Refacto answering

### PR DESCRIPTION
**Pull request by chrisdugne**
**Detailed purpose of the PR**

**Result and observation**

- [ ] **Breaking changes ?**  
       If checked, what have you broken ?

- [ ] **Extra lib ?**
      If checked, Which extra lib did you add ? (name, purpose, link ...).

**Testing Strategy**

- [ ] Already covered by tests
- [ ] Manual testing
- [x] Unit testing

### CODE REVIEW

**Comment by chrisdugne** [comment]
[ETQdev jai validateAnswer/editAnswer implementé coté store et simplifié pour clients web/mobile](https://trello.com/c/H5rl0r0D/1089-etqdev-jai-validateanswer-editanswer-implement%C3%A9-cot%C3%A9-store-et-simplifi%C3%A9-pour-clients-web-mobile)

File: packages/@coorpacademy-player-web/src/map-state-to-props/answer.js [hunk 1]
```diff
@@ -16,23 +16,10 @@ import isNil from 'lodash/fp/isNil';
...
 const editAnswerAction = (options, {dispatch}) => (state, slide) => newValue => {
-  return dispatch(
-    editAnswer(
-      getAnswerValues(slide, state),
-      getQuestionType(slide),
-      getCurrentProgressionId(state),
-      newValue
-    )
-  );
+  return dispatch(editAnswer(newValue));
```
Comment (chrisdugne) [inline packages/@coorpacademy-player-web/src/map-state-to-props/answer.js]: goal  1

File: packages/@coorpacademy-player-store/src/actions/ui/answers.js [hunk 2]
```diff
@@ -65,34 +67,61 @@ const newState = (
...
+): DispatchedAction => {
+  const state = getState();
+  const slide = getCurrentSlide(state);
+
+  if (!slide) {
+    return dispatch({
+      type: EDIT_ANSWER_ERROR,
+      meta: 'Cannot edit this answer, slide is not found'
+    });
+  }
```
Comment (godu) [inline packages/@coorpacademy-player-store/src/actions/ui/answers.js]: Pourquoi c'est pas dans un reducer ?

Comment (chrisdugne) [reply packages/@coorpacademy-player-store/src/actions/ui/answers.js]: pour devenir
```
getCurrentSlide = State => Slide
```
au lieu de
```
getCurrentSlide = State => Slide | void
```,

completement 👍 je note ça

Comment (chrisdugne) [reply packages/@coorpacademy-player-store/src/actions/ui/answers.js]: https://trello.com/c/sCHt0LV2/325-bouger-les-warnings-dans-les-reducers-plutot-que-dans-les-actioncreators

File: packages/@coorpacademy-player-web/src/map-state-to-props/player.js [hunk 3]
```diff
@@ -65,13 +64,7 @@ const playerProps = (options, store) => state => {
...
   const clickSeeClueHandler = () => dispatch(getClue);
   const clickBackToAnswerHandler = () => dispatch(selectRoute('answer'));
-  const clickCTAHandler = () =>
-    dispatch(
-      validateAnswer(getCurrentProgressionId(state), {
-        answer: getAnswerValues(slide, state),
-        slideId: slide._id
-      })
-    );
+  const clickCTAHandler = () => dispatch(validateAnswer());
```
Comment (chrisdugne) [inline packages/@coorpacademy-player-web/src/map-state-to-props/player.js]: goal 2

File: packages/@coorpacademy-player-store/src/utils/state-extract.js [hunk 4]
```diff
@@ -148,6 +149,16 @@ export const getLevel = (ref: string): (State => Level | void) => (state: State)
   state.data.contents.level.entities &&
   state.data.contents.level.entities[ref];
 
+export const getDiscipline = (ref: string): (State => Discipline | void) => (
```
Comment (chrisdugne) [inline packages/@coorpacademy-player-store/src/utils/state-extract.js]: goal 3

**Pull request merged**